CMakeLists.txt

```diff
@@ -651,35 +652,67 @@ if(Iconv_FOUND)
 	target_link_libraries(torrent-rasterbar PRIVATE Iconv::Iconv)
 endif()
 
-find_public_dependency(OpenSSL)
-set_package_properties(OpenSSL
+find_public_dependency(GnuTLS)
+set_package_properties(GnuTLS
 	PROPERTIES
-		URL "https://www.openssl.org/"
-		DESCRIPTION "Full-strength general purpose cryptography library"
+		URL "https://www.gnutls.org/"
+		DESCRIPTION "GnuTLS is a free software implementation of the TLS and DTLS protocols"
 		TYPE RECOMMENDED
 		PURPOSE "Provides HTTPS support to libtorrent"
 )
 
-if(OPENSSL_FOUND)
+if(NOT gnutls)
+	find_public_dependency(OpenSSL)
+	set_package_properties(OpenSSL
+		PROPERTIES
+			URL "https://www.openssl.org/"
+			DESCRIPTION "Full-strength general purpose cryptography library"
+			TYPE RECOMMENDED
+			PURPOSE "Provides HTTPS support to libtorrent"
+	)
+	if(OPENSSL_FOUND)
+		# TODO: needed until https://gitlab.kitware.com/cmake/cmake/issues/19263 is fixed
+		if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
+			target_link_libraries(OpenSSL::Crypto INTERFACE crypt32)
+		endif()
 
-	# TODO: needed until https://gitlab.kitware.com/cmake/cmake/issues/19263 is fixed
-	if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
-		target_link_libraries(OpenSSL::Crypto INTERFACE crypt32)
+		target_link_libraries(torrent-rasterbar PUBLIC OpenSSL::SSL)
+		target_compile_definitions(torrent-rasterbar
+			PUBLIC
+				TORRENT_USE_OPENSSL
+				TORRENT_SSL_PEERS
+				OPENSSL_NO_SSL2)
 	endif()
+endif()
 
-	target_link_libraries(torrent-rasterbar PUBLIC OpenSSL::SSL)
-	target_compile_definitions(torrent-rasterbar
-		PUBLIC
-			TORRENT_USE_LIBCRYPTO
-			TORRENT_USE_OPENSSL
-			TORRENT_SSL_PEERS
-			OPENSSL_NO_SSL2)
-	target_sources(torrent-rasterbar PRIVATE src/pe_crypto)
+if(gnutls OR NOT OPENSSL_FOUND)
+	find_public_dependency(GnuTLS)
+	set_package_properties(GnuTLS
+		PROPERTIES
+			URL "https://www.gnutls.org/"
+			DESCRIPTION "GnuTLS is a free software implementation of the TLS and DTLS protocols"
+			TYPE RECOMMENDED
+			PURPOSE "Provides HTTPS support to libtorrent"
+	)
+	if(GNUTLS_FOUND)
+		target_link_libraries(torrent-rasterbar PUBLIC GnuTLS::GnuTLS)
+		target_compile_definitions(torrent-rasterbar
+			PUBLIC
+				TORRENT_USE_GNUTLS
+				TORRENT_SSL_PEERS)
+		target_include_directories(torrent-rasterbar PUBLIC
+			$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/deps/asio-gnutls/include>
+            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+		install(DIRECTORY deps/asio-gnutls/include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+	elseif(gnutls)
+		message(FATAL_ERROR "GnuTLS library not found")
+	endif()
 endif()
 
 if(OPENSSL_FOUND)
 	target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_USE_LIBCRYPTO)
-else(OPENSSL_FOUND)
+	target_sources(torrent-rasterbar PRIVATE src/pe_crypto)
```

Comment:
`pe_crypto` implements the bittorrent "encryption"/obfuscation protocol. It uses diffie hellman and RC4. It doesn't require libcrypto, but I think it will use its routines if it's available. It's already being added below, if `encryption` is enabled. I don't think it needs to be added here, or in the else-clause